### PR TITLE
[it] tutorial-keras-text_classification

### PR DESCRIPTION
**Pull request by codeadmin-peritiae**
Basic translation

### CODE REVIEW

**Comment by codeadmin-peritiae** [comment]
Dear @lamberta , I checked the notebook tutorial with nbfmt and I've executed it in CoLab as well. 
No explicit issue or error.
I really don't understand the above automatic check failure...
Have you some suggestion? 
Thank you
Gianluca

**Comment by bhack** [comment]
Please check https://github.com/tensorflow/docs-l10n/blob/master/site/it/README.md#guida-alla-traduzione-italiana

You don't need to translate some words.

**Comment by codeadmin-peritiae** [comment]
> Please check https://github.com/tensorflow/docs-l10n/blob/master/site/it/README.md#guida-alla-traduzione-italiana
> 
> You don't need to translate some words.

To @bhack 
I reviewed the translation.
If some word is translated, despite his presence in the README file, is because, in the context where it is used, it has no specific technical meaning. 
For example, sometime the word "batch" is translated into "lotto" where it colloquially means "a bunch of data"; on the other side, the same word is not translated when it identify a specific type of background processing like"a batch process..." 
So, can you be more specific, please... Which words are you referring to?
Thank you.

**Comment by bhack** [comment]
Words like `layers` and others have a common usage "as Is" in the deep learning domain also in Italian.

**Comment by bhack** [comment]
Also `batch` is specific in Deep Learning and If you translate it in lotto Is confusing IMHO.
I don't think that you will talk about `lotto` between Deep Learning Italian researchers/engineers.

**Comment by codeadmin-peritiae** [comment]
> Words like `layers` and others have a common usage "as Is" in the deep learning domain also in Italian.
Just a couple of comments.
I intended these notebooks as disclosure documents that have to be able to be understandable to absolute beginners.
The proliferation of foreign terms in an Italian text (a research document too) ends up making the text incomprehensible; if not to insiders.
Because of my 25 years long experience in Italian ICT, I know very well the Italian (technology) people's inclination to use foreign terms, even if there are perfectly corresponding Italian words.
The meaning of "layers" in ML is the usual meaning we can found in any other "engineering" vocabulary: it identifies a stratum with some structural/filtering/processing functionality.
These are the main reasons because about "layer" (and "batch" too) I suggest to use Italian words.
In any case this is just my personal opinion, and if you wanna update my translation, you are absolutely free to do it.

**Comment by bhack** [comment]
I understand this point of view but then in the first Italian tutorial, blog or just chatting  with co-workers or other student you will be exposed to  layers, loss, batch etc and you will not understand what they are talking about.
It Is like to be exposed only to `elaboratore` and then somebody colloquial talk to you about PC or computer and you don't know what he Is talking about. 

Then every language has its own lingustic and cultural factors about the relationship with foreign words so not all languages have the same relationship.

I think that we could use a glossary that Is required for beginner to expose basic terms something like https://developers.google.com/machine-learning/glossary @lamberta What do you think?

**Comment by codeadmin-peritiae** [comment]
> I understand this point of view but then in the first Italian tutorial, blog or just chatting with co-workers or other student you will be exposed to layers, loss, batch etc and you will not understand what they are talking about.
> It Is like to be exposed only to `elaboratore` and then somebody colloquial talk to you about PC or computer and you don't know what he Is talking about.
> 
> Then every language has its own linguistic and cultural factors about the relationship with foreign words so not all languages have the same relationship.
> 
> I think that we could use a glossary that Is required for beginner to expose basic terms something like https://developers.google.com/machine-learning/glossary @lamberta What do you think?

Leaving an untranslated term into a text, we are leaving a sort of "symbol", but the more symbols you put in a text, the less readable it will be. 

Glossary is a good idea; we need it with English terms and Italian definitions. (I don't see it into docs-l10n).
In the notebooks we could use the Italian translation followed by the English term (on brackets) linked  to the glossary.

What do you (and @lamberta) think about?

**Comment by bhack** [comment]
Any other reviewer can make a pass over this? So that we could merge.

**Comment by mcaci** [comment]
I have had a quick pass over the text and the translation, the translation and the comments are good and very interesting for me. In my personal experience I am not used to deal more with and use English terms rather than translated ones.

In the examples of the comments the word "lotto" feels a bit unnatural to me compared to "batch", the word "livello" rather than "layer" feels a bit more natural, but still I would lean over the untranslated word for such technical concepts. This eventually is my 2 cents.

I agree that for those who might have a bit more difficulty with meaning of these untranslated terms a glossary would make sense, but I would wait feedback from the Italian readers to push this proposal in order to see how they already deal with the translation. What do you think?

Other than that the translation overall looks good and the comments bring up very good points.

**Comment by bhack** [comment]
@mcaci Yes I belive that a good intro glossary is the best balance. But I that is an infra for all the languages so we need to wait for @lamberta.

**Comment by lamberta** [comment]
Hey all, sorry for the delay.

Regarding the CI failure, this is a new formatting check that all notebooks must pass. By using a common format for the notebook JSON, it means our diffs will be easier to read on subsequent changes (notebooks in source control is a challenge). It also helps with "diff churn" when different notebook editors add their own metadata fields, etc.

You just need to run your notebook through the [nbfmt.py](https://github.com/tensorflow/docs/blob/master/tools/nbfmt.py) tool:

```
# Install deps:
$ pip3 install -U [--user] absl-py

# Assume you've cloned a local version of the tensorflow/docs repo ...
$ ../docs/tools/nbfmt.py site/it/your/notebook.ipynb
```

We'll be adding more lint checks to the GitHub CI in the coming weeks since managing all these notebooks without automation has become quite the challenge :)

**Comment by lamberta** [comment]
The GitHub message indicated the problem was with the `text_classification.ipynb` notebook. I pulled your PR locally, ran `nbfmt.py` on it, and pushed the commit to your PR branch. The check now passes so looks like it's working-as-intended :)

File: site/it/tutorials/keras/text_classification.ipynb [hunk 1]
```diff
@@ -0,0 +1,729 @@
...
+      "source": [
+        "<table class=\"tfo-notebook-buttons\" align=\"left\">\n",
+        "  <td>\n",
+        "    <a target=\"_blank\" href=\"https://www.tensorflow.org/tutorials/keras/text_classification\"><img src=\"https://www.tensorflow.org/images/tf_logo_32px.png\" />Visualizza su TensorFlow.org</a>\n",
+        "  </td>\n",
+        "  <td>\n",
+        "    <a target=\"_blank\" href=\"https://colab.research.google.com/github/codeadmin-peritiae/docs-l10n/blob/tutorial-keras-text_classification/site/it/tutorials/keras/text_classification.ipynb\"><img src=\"https://www.tensorflow.org/images/colab_logo_32px.png\" />Esegui in Google Colab</a>\n",
+        "  </td>\n",
+        "  <td>\n",
+        "    <a target=\"_blank\" href=\"https://github.com/codeadmin-peritiae/docs-l10n/blob/tutorial-keras-text_classification/site/it/tutorials/keras/text_classification.ipynb\"><img src=\"https://www.tensorflow.org/images/GitHub-Mark-32px.png\" />Visualizza il sorgente su GitHub</a>\n",
```
Comment (lamberta) [inline site/it/tutorials/keras/text_classification.ipynb:96]: These links should point to the tensorflow/docs-l10n master branch

File: site/it/tutorials/keras/text_classification.ipynb [hunk 2]
```diff
@@ -0,0 +1,729 @@
...
+      "cell_type": "code",
+      "execution_count": 0,
+      "metadata": {
+        "colab": {},
+        "colab_type": "code",
+        "id": "Nh0KjNGMWNlL"
+      },
+      "outputs": [],
+      "source": [
+        "from __future__ import absolute_import, division, print_function, unicode_literals"
```
Comment (lamberta) [inline site/it/tutorials/keras/text_classification.ipynb:151]: removed in latest version of notebook since we're only python 3 now

File: site/it/tutorials/keras/text_classification.ipynb [hunk 3]
```diff
@@ -0,0 +1,729 @@
...
+        "colab_type": "code",
+        "id": "8RZOuS9LWQvv"
+      },
+      "outputs": [],
+      "source": [
+        "try:\n",
+        "  # %tensorflow_version only exists in Colab.\n",
+        "  !pip install tf-nightly\n",
+        "except Exception:\n",
+        "  pass\n",
```
Comment (lamberta) [inline site/it/tutorials/keras/text_classification.ipynb:168]: removed in latest version of notebook since Colab now defaults to TF2.

(You may want to give another quick look at the source en/ version to make sure this is up-to-date.)

**Comment by lamberta** [comment]
Ok, catching up on the thread a bit ... I can't help with the Italian but I can provide some guidance based on the how the other language communities have handled similiar problems, an provide some background.

So we started translating tensorflow.org technical docs using vendors sourced by an internal team. Unfortunately, based on community feedback, the translation quality was deemed "insufficient" for machine learning technical documentation. A common pattern I saw was *over-translating* technical terms that had a precise meaning into something unclear for developers. One notable quote that I find myself saying to this day is: [in Russian accent] "WTF is mini-batch in Russian?!"

So odd translations, and the vendor's inability to work with Jupyter notebooks in GitHub led to the community taking matters into their own hands.

It's not that the vendor translations are bad, it's just very difficult to find translators with a developer background and a particular interest in machine learning. And writing translations is a difficult task that's under-appreciated for the technical and creative skills required. But also very rewarding because you're giving lots and lots of people access to material they otherwise would have a difficult time going through, so thanks! 😃

So to that end, consistency is good but I don't think it should be a blocker. Agreement on common terms and style is important, but it would be good to decide on a process for this that doesn't leave contributions in limbo—whatever works for the community. And based on other PRs, it looks like you've been doing this. It's GitHub so you can always edit and search/replace docs later.

Hope that helps!

File: site/it/tutorials/keras/text_classification.ipynb (hunk 1 above)
Comment (codeadmin-peritiae) [reply site/it/tutorials/keras/text_classification.ipynb:96]: Ok

File: site/it/tutorials/keras/text_classification.ipynb (hunk 2 above)
Comment (codeadmin-peritiae) [reply site/it/tutorials/keras/text_classification.ipynb:151]: Ok. Thank you

File: site/it/tutorials/keras/text_classification.ipynb (hunk 3 above)
Comment (codeadmin-peritiae) [reply site/it/tutorials/keras/text_classification.ipynb:168]: Done

**Comment by codeadmin-peritiae** [comment]
Switch to different repository
